vertica_python/vertica/cursor.py

```diff
@@ -83,7 +82,9 @@ def fetchone(self):
             self._message = self.connection.read_message()
             return row
         else:
-            self.connection.process_message(self._message)
+            while not isinstance(self._message, messages.ReadyForQuery):
```

Comment:
If a ReadyForQuery is received, we should clear out the existing execution and capture the transaction status.